signer/Config.go

```diff
@@ -11,36 +16,231 @@ import (
 	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
 	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
 	"github.com/cosmos/cosmos-sdk/types/bech32"
+	"github.com/strangelove-ventures/horcrux/client"
 	"github.com/tendermint/tendermint/crypto"
+	"gopkg.in/yaml.v2"
 )
 
 type NodeConfig struct {
 	Address string
 }
 
+// Config maps to the on-disk JSON format
+type DiskConfig struct {
+	PrivValKeyFile *string         `json:"key-file,omitempty" yaml:"key-file,omitempty"`
+	ChainID        string          `json:"chain-id" yaml:"chain-id"`
+	CosignerConfig *CosignerConfig `json:"cosigner,omitempty" yaml:"cosigner,omitempty"`
+	ChainNodes     ChainNodes      `json:"chain-nodes,omitempty" yaml:"chain-nodes,omitempty"`
+	DebugAddr      string          `json:"debug-addr,omitempty" yaml:"debug-addr,omitempty"`
+}
+
+func (c *DiskConfig) Nodes() (out []string) {
+	for _, n := range c.ChainNodes {
+		out = append(out, n.PrivValAddr)
+	}
+	return out
+}
+
+func (c *DiskConfig) MustMarshalYaml() []byte {
+	out, err := yaml.Marshal(c)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
+
+func (c *DiskConfig) CosignerPeers() (out []CosignerParams) {
+	for _, p := range c.CosignerConfig.Peers {
+		out = append(out, CosignerParams{ID: p.ShareID, Address: p.P2PAddr})
+	}
+	return
```

Comment:
Nit: In Go, many think it's best avoid naked returns even if it shortens LOC. [Issue](https://github.com/golang/go/issues/21291). 

I don't like them because of the indirection. I have to now traverse the function to figure out what, exactly, is being returned.

Reply:
definitely, I missed this one. updated.